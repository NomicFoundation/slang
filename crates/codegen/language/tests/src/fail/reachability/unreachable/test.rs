#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = Bar1,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0", "2.0.0", "3.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                // reachable (from root)
                Struct(name = Bar1, fields = (field = Required(NonTerminal(Bar2)))),
                Struct(name = Bar2, fields = (field = Required(NonTerminal(Bar1)))),
                // not-reachable
                Struct(name = Baz1, fields = (field = Required(NonTerminal(Baz2)))),
                Struct(name = Baz2, fields = (field = Required(NonTerminal(Baz1))))
            ]
        )]
    )]
));

fn main() {}
