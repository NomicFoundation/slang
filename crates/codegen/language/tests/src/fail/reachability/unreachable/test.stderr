error: Item 'Baz1' is not reachable from grammar root.
  --> src/fail/reachability/unreachable/test.rs:20:31
   |
20 |                 Struct(name = Baz1, fields = (field = Required(Baz2))),
   |                               ^^^^

error: Item 'Baz2' is not reachable from grammar root.
  --> src/fail/reachability/unreachable/test.rs:21:31
   |
21 |                 Struct(name = Baz2, fields = (field = Required(Baz1)))
   |                               ^^^^
